### PR TITLE
Add a Tooltip demo

### PR DESCRIPTION
**Pull request by larsiusprime**
Add Tooltips demo to flixel-demos

### CODE REVIEW

File: User Interface/Tooltips/assets/xml/state_code.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<data>
+	
+	<!--blank-->
```
Comment (Gama11) [inline User Interface/Tooltips/assets/xml/state_code.xml]: Why is this needed? Can't you just not specify any xml?

File: User Interface/Tooltips/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+Showcases the basic widgets in [flixel-ui](https://github.com/HaxeFlixel/flixel-ui).
```
Comment (Gama11) [inline User Interface/Tooltips/README.md]: This seems to be the readme from RPG Interface?

File: User Interface/Tooltips/.gitignore [hunk 3]
```diff
@@ -0,0 +1,5 @@
+/flixel/
```
Comment (Gama11) [inline User Interface/Tooltips/.gitignore]: Do you need this .gitignore for anything? export folders are already ignored in the gitignore in the root directory. Same goes for RPG Interface I guess, since it seems to have been copied from there.

File: User Interface/Tooltips/source/State_DemoCode.hx [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+				}
+		}
+	}
+	
+	private function makeByHand():Void
+	{
+		_ui.addAsset(cast new FlxUISprite().makeGraphic(FlxG.width, FlxG.height, 0xFF404040), "bkg");
+		
+		var border  = new BorderDef(FlxTextBorderStyle.OUTLINE, FlxColor.BLACK, 1, 1);
+		var sans8   = new FontDef("vera", ".ttf", "assets/fonts/verab.ttf", new TextFormat(null, 8, FlxColor.WHITE, true));
```
Comment (Gama11) [inline User Interface/Tooltips/source/State_DemoCode.hx]: Seeing how the font size is almost the only thing varying here, you could have a `createFontDef()` helper function to get rid of the redundancy.

File: User Interface/RPG Interface/source/State_Title.hx [hunk 5]
```diff
@@ -66,6 +66,15 @@ class State_Title extends FlxUIState
 		}		
 	}	
 	
+	public override function update(elapsed:Float):Void
```
Comment (Gama11) [inline User Interface/RPG Interface/source/State_Title.hx]: Isn't this the same debug code as in #196? :)

File: User Interface/Tooltips/assets/xml/state_code.xml (hunk 1 above)
Comment (larsiusprime) [inline User Interface/Tooltips/assets/xml/state_code.xml]: If you don't specify an xml, it doesn't create the _ui object. Let me fix that with a quick edit...

File: User Interface/Tooltips/README.md (hunk 2 above)
Comment (larsiusprime) [inline User Interface/Tooltips/README.md]: Yup... copypasta error.

File: User Interface/Tooltips/source/State_DemoCode.hx (hunk 4 above)
Comment (larsiusprime) [inline User Interface/Tooltips/source/State_DemoCode.hx]: Ok

File: User Interface/RPG Interface/source/State_Title.hx (hunk 5 above)
Comment (larsiusprime) [inline User Interface/RPG Interface/source/State_Title.hx]: fixed, among other things

**Comment by larsiusprime** [comment]
Is this ready to merge yet? I think I'm done with the other one:
https://github.com/HaxeFlixel/flixel-ui/pull/114

**Comment by larsiusprime** [comment]
All the dependent PR's are merged in now.

**Comment by MSGhero** [comment]
This should be merged and posted to the site. The failed checks are only 3.1.3.

**Comment by larsiusprime** [comment]
This seems fine to me? Okay to merge/post this @Gama11 ?

**Comment by Gama11** [comment]
Didn't realize this was still open. Should probably drop 3.1.3 compatiblity first to avoid travis failures.

@MSGhero we don't put dev-branch-only demos on the website right now.
